### PR TITLE
Add codecov bot.

### PR DESCRIPTION
**Pull request by MrLYG**
Add codecov bot to comment on the current code coverage under each pr committed to the `trainers `and `abcd` branches.
### Instruction
1. Register for Codecov:
First, visit [Codecov](https://codecov.io/) and register for a new account. You can register directly with your GitHub account.

2. Import Your Project:
After registration, you will be redirected to your Codecov dashboard. Click on the "Add new repository" button on the left side, then select your GitHub repository to import on the following page.

3. Generate a Codecov Token and Add to GitHub:
If this is your first time importing a project, you will see several instructional steps, you only need to complete the first two (just complete the two steps in the image below)
image

### Notes
After the bot is added, it may not immediately display its effect. The bot functions by comparing the base commit of the current PR with the PR itself. When we initially added this bot, we did not incorporate the code coverage report of the trainers or abcd branch code into Codecov. Therefore, the pertinent coverage will not appear here.

However, after this PR is merged into the `trainers` branch, all subsequent branches that are checked out from the trainers branch commit will display the relevant code coverage in the comments under the changed PR. If desired, these comments can be turned off.

It's also important to note that code coverage reports for all commits can be accessed in the Codecov dashboard.

### CODE REVIEW

**Comment by fostiropoulos** [comment]
@MrLYG Can we please update the status on this. Would this be ready to be merged?

File: .github/workflows/codecov_report_submit.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name:  Codecov CI
+
+on:
+  push:
+    branches: [ trainers, abcd, ]
+  pull_request:
+    branches: [ trainers, abcd, ]
```
Comment (MrLYG) [inline .github/workflows/codecov_report_submit.yml]:
```suggestion
on:
  push:
    branches: trainers
  pull_request:
    branches: trainers
```
> @MrLYG Can we please update the status on this? Would this be ready to be merged?

@fostiropoulos 
Totally, yes.
And before the merge, I want to ask:
I remember you said to me that the `trainers` branch become the main branch we need to work on.
So, is it mean the `abcd` branch is deprecated? if it is, I will change the code here. That's mean we only focus on `abcd` trainers. Any push and pr submit to `abcd` will never trigger code-coverage to generate a report.

Comment (fostiropoulos) [reply .github/workflows/codecov_report_submit.yml]: sounds good

Comment (MrLYG) [reply .github/workflows/codecov_report_submit.yml]: > sounds good

Ready to merge.

**Review by fostiropoulos — APPROVED** [review]

**Pull request merged**